sqlx-core/src/connection.rs

```diff
@@ -33,6 +33,19 @@ pub trait Connection: Send {
     ///
     /// Returns a [`Transaction`] for controlling and tracking the new transaction.
     fn begin(&mut self) -> BoxFuture<'_, Result<Transaction<'_, Self::Database>, Error>>
+    where
+        Self: Sized,
+    {
+        Self::begin_with(self, Default::default())
```

Comment:
Why not take the same approach for the other impls  (like `impl Acquire for Pool`)?